### PR TITLE
[C++] Fix coverity issues

### PR DESCRIPTION
**Pull request by PiotrKrzem**
### Details:
 - Fixed ostream not restoring previous format due to the incorrect use of std::cout
 - Fixed AUTO_CAUSES_COPY error due to the missing '&' in for loops in the main file
### Tickets:
 - 100701

### CODE REVIEW

File: samples/cpp/benchmark_app/main.cpp [hunk 1]
```diff
@@ -437,7 +437,7 @@ int main(int argc, char* argv[]) {
                             std::stringstream strm(it_device_nstreams->second);
                             std::map<std::string, std::string> devices_property;
                             ov::util::Read<std::map<std::string, std::string>>{}(strm, devices_property);
-                            for (auto it : devices_property) {
+                            for (auto& it : devices_property) {
```
Comment (ilyachur) [inline samples/cpp/benchmark_app/main.cpp:440]:
```suggestion
                            for (const auto& it : devices_property) {
```
You don't change `it`

File: samples/cpp/benchmark_app/main.cpp [hunk 2]
```diff
@@ -513,7 +513,7 @@ int main(int argc, char* argv[]) {
                             std::stringstream strm(it_device_infer_precision->second);
                             std::map<std::string, std::string> devices_property;
                             ov::util::Read<std::map<std::string, std::string>>{}(strm, devices_property);
-                            for (auto it : devices_property) {
+                            for (auto& it : devices_property) {
```
Comment (ilyachur) [inline samples/cpp/benchmark_app/main.cpp:516]: The same

Comment (PiotrKrzem) [reply samples/cpp/benchmark_app/main.cpp]: Applied!

**Review by Wovchena — APPROVED** [review]

**Review by t-jankowski — APPROVED** [review]

**Comment by Wovchena** [comment]
@PiotrKrzem, please restart the build. I didn't figure out how to restart pot manually, you may need to restart it by committing to the branch

**Comment by PiotrKrzem** [comment]
> @PiotrKrzem, please restart the build. I didn't figure out how to restart pot manually, you may need to restart it by committing to the branch

Done - Although I did have to merge with master due to the version mismatch in wheel installation

**Pull request merged**